src/core/syscalls/business_logic_syscall_handler.rs

```diff
@@ -269,78 +296,67 @@ where
         vm: &VirtualMachine,
         syscall_ptr: Relocatable,
     ) -> Result<Vec<Felt>, SyscallHandlerError> {
-        // Parse request and prepare the call.
-        let request =
-            match self._read_and_validate_syscall_request(syscall_name, vm, syscall_ptr)? {
-                SyscallRequest::CallContract(request) => request,
-                _ => return Err(SyscallHandlerError::ExpectedCallContract),
-            };
-
-        let mut class_hash = None;
-        let calldata = get_integer_range(vm, &request.calldata, request.calldata_size)?;
+        let request = self._read_and_validate_syscall_request(syscall_name, vm, syscall_ptr)?;
 
+        let entry_point_type;
+        let function_selector;
+        let class_hash;
         let contract_address;
         let caller_address;
-        let entry_point_type;
         let call_type;
-        match syscall_name {
-            "call_contract" => {
-                contract_address = request.contract_address;
-                caller_address = self.contract_address.clone();
-                entry_point_type = EntryPointType::External;
-                call_type = CallType::Call;
-            }
-            "delegate_call" => {
-                contract_address = self.contract_address.clone();
-                caller_address = self.caller_address.clone();
-                entry_point_type = EntryPointType::External;
-                call_type = CallType::Delegate;
-            }
-            "delegate_l1_handler" => {
-                contract_address = self.contract_address.clone();
-                caller_address = self.caller_address.clone();
-                entry_point_type = EntryPointType::L1Handler;
-                call_type = CallType::Delegate;
-            }
```

Comment:
why were these cases deleted?

Reply:
I rewrote the entire function since it wasn't working as intended. I omitted the (for now) unused branches for simplicity.